### PR TITLE
Support for canceling execution plans (inluding scheduled and plans with sub-plans)

### PR DESCRIPTION
**Pull request by iNecas**

### CODE REVIEW

**Comment by iNecas** [comment]
@adamruzicka could you review it please?

File: lib/dynflow/scheduled_plan.rb [hunk 1]
```diff
@@ -34,6 +34,12 @@ def error(message, history_entry = nil)
       execution_plan.root_plan_step.save
       execution_plan.execution_history.add history_entry, @world.id unless history_entry.nil?
       execution_plan.update_state :stopped
+      @world.persistence.delete_scheduled_plans(:execution_plan_uuid => execution_plan.id)
```
Comment (adamruzicka) [inline lib/dynflow/scheduled_plan.rb]: I would suggest moving this line to the `cancel` method (or making additional changes to scheduler behavior).

When a scheduler finds an expired schedule record, it calls `error` on its execution plan and deletes the schedule record afterwards (along with other processed schedule records). This way expired schedule records would end up being deleted twice.

File: lib/dynflow/scheduled_plan.rb [hunk 2]
```diff
@@ -34,6 +34,12 @@ def error(message, history_entry = nil)
...
+    end
+
+    def cancel
+      error("Scheudled task cancelled", "Scheduled task cancelled")
```
Comment (adamruzicka) [inline lib/dynflow/scheduled_plan.rb]: Typo here

File: lib/dynflow/scheduled_plan.rb (hunk 1 above)
Comment (iNecas) [inline lib/dynflow/scheduled_plan.rb]: Updated

File: lib/dynflow/scheduled_plan.rb (hunk 2 above)
Comment (iNecas) [inline lib/dynflow/scheduled_plan.rb]: fixed

**Comment by adamruzicka** [comment]
No more comments, ACK

**Comment by iNecas** [comment]
Thanks for review @adamruzicka

**Pull request merged**